examples/pipeline_wavernn/datasets.py

```diff
@@ -0,0 +1,114 @@
+import os
+import random
+import torch
+import torchaudio
+from torchaudio.datasets import LJSPEECH
+
+
+class ProcessedLJSPEECH(LJSPEECH):
+
+    def __init__(self, files, transforms, mode, mulaw, n_bits):
+
+        self.transforms = transforms
+        self.files = files
+        self.mulaw = mulaw
+        self.n_bits = n_bits
+        self.mode = mode
+
+    def __getitem__(self, index):
+
+        file = self.files[index]
+
+    # use torchaudio transform to get waveform and specgram
+        # waveform, sample_rate = torchaudio.load(file)
+        # specgram = self.transforms(x)
+        # return waveform.squeeze(0), mel.squeeze(0)
+
+    # use librosa transform to get waveform and specgram
+        waveform = self.transforms.load(file)
+        specgram = self.transforms.melspectrogram(waveform)
+
+        # waveform and spectrogram: [0, 2**bits-1].
+        if self.mode == 'waveform':
+            waveform = self.transforms.mulaw_encode(waveform, 2**self.n_bits) if self.mulaw \
+                else float_2_int(waveform, self.n_bits)
+
+        elif self.mode == 'mol':
+            waveform = float_2_int(waveform, 16)
+
+        return waveform, specgram
+
+    def __len__(self):
+        return len(self.files)
+
+
+# From float waveform [-1, 1] to integer label [0, 2 ** bits - 1]
+def float_2_int(waveform, bits):
```

Comment:
Please add a standard docstring describing what this is doing, instead of the comment above.

Reply:
Good suggestion.
- I added the description in the function docstring.
- The name is changed to `waveform_to_label`.
- I put this function in the `utils` file.